src/Token/Parser.php

```diff
@@ -105,7 +107,19 @@ private function parseClaims(string $data): array
                 continue;
             }
 
-            $claims[$claim] = $this->convertDate((string) $claims[$claim]);
+            $claimvalue = $claims[$claim];
+
+            if (! is_string($claimvalue)) {
+                $encoded = json_encode($claimvalue);
+
+                if ($encoded !== false) {
+                    $claims[$claim] = $this->convertDate($encoded);
+
+                    continue;
+                }
+            }
+
+            $claims[$claim] = $this->convertDate((string) $claimvalue);
```

Comment:
We shouldn't have issues encoding this again to json, so this can be simplified to:

```suggestion
            $date = $claims[$claim];

            $claims[$claim] = $this->convertDate(is_string($date) ? $date : json_encode($date, JSON_THROW_ON_ERROR));
```